components/server/src/server.ts

```diff
@@ -340,11 +348,20 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
             });
         }
 
+        if (this.iamSessionApp) {
```

Comment:
What's the reasoning for starting a new server (on another port), rather than using an existing one?

Reply:
This pattern is used to have separate ingress rules. 
* port 3000 is the main app
* port 9000 is the installation admin
* ...
* port 9876 (randomly chosen to avoid conflicts) is the internal handler for `iam`

Maybe that's a bit paranoid, and we alternatively we could add proper authZ, but that way we can make sure, that this skateboard isn't an open door.